sssom/parsers.py

```diff
@@ -590,41 +591,44 @@ def to_mapping_set_document(msdf: MappingSetDataFrame) -> MappingSetDocument:
     mlist = []
     ms = MappingSet()
     bad_attrs = {}
-    for _, row in msdf.df.iterrows():
-        mdict = {}
-        for k, v in row.items():
-            ok = False
-            if k:
-                k = str(k)
-            if hasattr(Mapping, k):
-                mdict[k] = v
-                ok = True
-            if hasattr(MappingSet, k):
-                ms[k] = v
-                ok = True
-            if not ok:
-                if k not in bad_attrs:
-                    bad_attrs[k] = 1
-                else:
-                    bad_attrs[k] += 1
-        m = _prepare_mapping(Mapping(**mdict))
-        mlist.append(m)
+    if msdf.df is not None:
+        for _, row in msdf.df.iterrows():
+            mdict = {}
```

Comment:
the code in the innermost loop should be a class method on `Mapping`

Reply:
Issue